server/src/updaters/thread-updaters.js

```diff
@@ -427,30 +429,65 @@ async function updateThread(
     nextThreadType !== oldThreadType ||
     nextParentThreadID !== oldParentThreadID
   ) {
-    savePromises.recalculatePermissionsChangeset = recalculateAllPermissions(
+    intermediatePromises.recalculatePermissionsChangeset = recalculateAllPermissions(
       request.threadID,
       nextThreadType,
     );
   }
   const {
     addMembersChangeset,
     recalculatePermissionsChangeset,
-  } = await promiseAll(savePromises);
+  } = await promiseAll(intermediatePromises);
 
-  const changeset = [];
+  const membershipRows = [];
+  const relationshipRows = [];
   if (recalculatePermissionsChangeset && newMemberIDs) {
-    changeset.push(
-      ...recalculatePermissionsChangeset.filter(
+    const {
+      membershipRows: recalculateMembershipRows,
+      relationshipRows: recalculateRelationshipRows,
+    } = recalculatePermissionsChangeset;
+    membershipRows.push(
+      ...recalculateMembershipRows.filter(
         rowToSave => !newMemberIDs.includes(rowToSave.userID),
       ),
     );
+    const parentMemberIDs = recalculateMembershipRows
+      .filter(rowToSave => !newMemberIDs.includes(rowToSave.userID))
+      .map(rowToSave => rowToSave.userID);
```

Comment:
Nit: I would flip the order of this `filter`/`map`, since we only care about `userID` anyways